domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,140 @@
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  private val mockPlatformParameterMap by lazy {
+    val stringPlatformParameter = ParameterValue.getDefaultInstance().toBuilder()
+      .setString(STRING_PLATFORM_PARAMETER_VALUE).build()
+    val integerPlatformParameter = ParameterValue.getDefaultInstance().toBuilder()
+      .setInteger(INTEGER_PLATFORM_PARAMETER_VALUE).build()
+    val booleanPlatformParameter = ParameterValue.getDefaultInstance().toBuilder()
+      .setBoolean(BOOLEAN_PLATFORM_PARAMETER_VALUE).build()
+
+    mapOf(
+      STRING_PLATFORM_PARAMETER_NAME to stringPlatformParameter,
+      INTEGER_PLATFORM_PARAMETER_NAME to integerPlatformParameter,
+      BOOLEAN_PLATFORM_PARAMETER_NAME to booleanPlatformParameter,
+    )
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testSingleton_initialPlatformParameterMap_isEmpty() {
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isEmpty()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_isNotEmpty() {
+    assertThat(platformParameterSingleton.getPlatformParameterMap().isEmpty()).isTrue()
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    assertThat(platformParameterSingleton.getPlatformParameterMap().isEmpty()).isFalse()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveStringParameter_verifyItsValue() {
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    val stringPlatformParameter = platformParameterSingleton.getStringPlatformParameter(
+      STRING_PLATFORM_PARAMETER_NAME
+    )
+    assertThat(stringPlatformParameter?.value).isEqualTo(STRING_PLATFORM_PARAMETER_VALUE)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveIntegerParameter_verifyItsValue() {
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    val integerPlatformParameter = platformParameterSingleton.getIntegerPlatformParameter(
+      INTEGER_PLATFORM_PARAMETER_NAME
+    )
+    assertThat(integerPlatformParameter?.value).isEqualTo(INTEGER_PLATFORM_PARAMETER_VALUE)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveBooleanParameter_verifyItsValue() {
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    val booleanPlatformParameter = platformParameterSingleton.getBooleanPlatformParameter(
+      BOOLEAN_PLATFORM_PARAMETER_NAME
+    )
+    assertThat(booleanPlatformParameter?.value).isEqualTo(BOOLEAN_PLATFORM_PARAMETER_VALUE)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveIncorrectParameter_verifyItsValue() {
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    val incorrectPlatformParameter = platformParameterSingleton.getStringPlatformParameter(
+      INCORRECT_PLATFORM_PARAMETER_NAME
+    )
+    assertThat(incorrectPlatformParameter).isNull()
+  }
```

Comment:
What happens when we do a getStringPlatformParameter but pass BOOLEAN_PLATFORM_PARAMETER_NAME to it? Can we add a test like this?

Reply:
it should return null. Added the test for it